activerecord/test/cases/relation/load_async_test.rb

```diff
@@ -41,6 +41,28 @@ def test_scheduled?
       assert_not_predicate deferred_posts, :scheduled?
     end
 
+    def test_async_pending_predicate_with_async_query
+      unless in_memory_db?
```

Comment:
If the test doesn't apply to `in_memory_db?` the unless should be outside the `def`, like other tests in this file.